Cython/Compiler/ExprNodes.py

```diff
@@ -9436,6 +9436,12 @@ def generate_cyfunction_code(self, code):
         if def_node.local_scope.parent_scope.is_c_class_scope and not def_node.entry.is_anonymous:
             flags.append('__Pyx_CYFUNCTION_CCLASS')
 
+        # mutually exclusive as far as "inspect" is concerned
+        if getattr(def_node, "is_asyncgen", None):
```

Comment:
Why would a simple attribute lookup not work here?

Reply:
Not all def_node have those attributes, they're added only on AsyncDefNode, and AsyncGenNode.

Reply:
Then we should add these attributes to `DefNode`. Because, why not?